### ISSUE TITLE
rowAnys, rowAlls, colAnys, colAlls have unexpected behaviour for numeric matrices

### ISSUE THREAD

**Issue by richierocks** [HenrikBengtsson/matrixStats#89]
For numeric matrices, I expected the rowwise and columnwise _any_ and _all_ functions to treat `0` as `FALSE` and non-zero values as `TRUE`.  (This is how base `any` and `all` work.)

``` r
library(matrixStats)
(m <- matrix(1.5 * 0:3, 2))
##      [,1] [,2]
## [1,]  0.0  3.0
## [2,]  1.5  4.5
rowAnys(m)
## [1] FALSE FALSE
rowAlls(m)
## [1] FALSE FALSE
colAnys(m)
## [1] FALSE FALSE
colAlls(m)
## [1] FALSE FALSE
```

Converting the contents of the matrix to be logical, I get the answer I expected.

``` r
m[] <- as.logical(m)
rowAnys(m)
## [1] TRUE TRUE
rowAlls(m)
## [1] FALSE  TRUE
colAnys(m)
## [1] TRUE TRUE
colAlls(m)
## [1] FALSE  TRUE
```

Is the behaviour with numeric matrices deliberate or a bug?  If it is the former, it really needs documenting, and possibly having a warning in the code.

**Comment by HenrikBengtsson** [comment]
Thanks for this.  I agree with you; it should be that:

* zero values corresponds to FALSE and
* non-zero values corresponds to TRUE.

However, the current behavior for numeric values is that:
* non-one values corresponds to FALSE and
* one values corresponds to TRUE.

Example:
```r
> x <- matrix(c(0,1/2,1,2), nrow=2)
> x
     [,1] [,2]
[1,]  0.0    1
[2,]  0.5    2
> rowAnys(x)
[1]  TRUE FALSE
> colAnys(x)
[1] FALSE  TRUE
```

The reason for this odd behavior is because these functions internally looks for the existance of a specific `value`, which defaults to `TRUE`, which in the numeric case is coerced to `1.0` via `as.numeric(TRUE)`.  This can be seen below:
```r
> rowAnys
function (x, rows = NULL, cols = NULL, value = TRUE, na.rm = FALSE, 
    dim. = dim(x), ...) 
{
    if (is.numeric(x) || is.logical(x)) {
        na.rm <- as.logical(na.rm)
        hasNAs <- TRUE
        counts <- .Call(C_rowCounts, x, dim., rows, cols, value, 
            1L, na.rm, hasNAs)
        as.logical(counts)
    }
[...]
```
Regardless of why, I agree with you - I'll fix this in the next release.

**Comment by HenrikBengtsson** [comment]
This is now (finally) fixed (commit dd3c62a) - it's in the `develop` branch for now.  Thanks for reporting.
